Document using DOCC_EXEC in SwiftPM (#653)
apple/swift-docc: README.md

# Swift-DocC

> Swift-DocC is a documentation compiler for Swift frameworks and packages aimed 

Author: Thank you for pointing it out — I didn't know about that. I've removed the whole formatting commit — it feels like I don't know enough about internal systems to make it warranted and safe, and I don't feel there's enough value for beginner engineers reading the docs, or for the core team, to justify spending your time on reviewing it further.

button in Xcode's Product menu, your custom `docc` will be used for the build.
You can confirm that your custom `docc` is being used by opening the latest build
log in Xcode's report navigator and expanding the "Compile documentation" step.

### Invoking `docc` from Swift Package Manager

You can also test a locally built version of Swift-DocC using Swift Package Manager

> CLI. SwiftPM will try to read `DOCC_EXEC` environment variable value, and use

Reviewer: I think it would be good to clarify here that isn't SwiftPM reading the `DOCC_EXEC` environment variable, that's a behavior of the Swift-DocC Plugin.

```suggestion
CLI. The Swift-DocC SwiftPM plugin will try to read `DOCC_EXEC` environment variable value, and use
```

> You can also test a locally built version of Swift-DocC using Swift Package Manager
> CLI. SwiftPM will try to read `DOCC_EXEC` environment variable value, and use

Reviewer: What do you think of avoiding the CLI abbreviation here? I think it's possible some reader's won't be familiar with it.

```suggestion
You can also test a locally built version of Swift-DocC using the Swift Package Manager from the command line.
SwiftPM will try to read `DOCC_EXEC` environment variable value, and use
```

the path you provded if it's set.

>   1. In your test `Package.swift`, add a dependency on [`Swift-DocC Plugin`](https://github.com/apple/swift-docc-plugin).

Reviewer: ```suggestion
  1. In your project's `Package.swift`, add a dependency on the [`Swift-DocC Plugin`](https://github.com/apple/swift-docc-plugin).
```

>   2. Set `DOCC_EXEC` environment variable and run documentation generation
>      command like this:

Reviewer: ```suggestion
  2. Set the `DOCC_EXEC` environment variable and run the documentation generation
     command:
```

Author: 🤦‍♂️

>         `DOCC_EXEC=/path/to/docc swift package generate-documentation`

Reviewer: What do you think of updating this to a code block to match the existing formatting? Currently it stands out from the others and is more difficult to copy/paste.

```suggestion
        ```bash
        DOCC_EXEC=/path/to/docc swift package generate-documentation
        ```
```

<img width="1112" alt="Screenshot 2023-07-07 at 10 39 25 AM" src="https://github.com/apple/swift-docc/assets/6750147/32c64d53-06f3-42bc-9b6b-69da7ad9d91f">

button in Xcode's Product menu, your custom `docc` will be used for the build.
You can confirm that your custom `docc` is being used by opening the latest build
log in Xcode's report navigator and expanding the "Compile documentation" step.

### Invoking `docc` from Swift Package Manager

You can also test a locally built version of Swift-DocC using Swift Package Manager
CLI. SwiftPM will try to read `DOCC_EXEC` environment variable value, and use
the path you provded if it's set.

>   1. In your test `Package.swift`, add a dependency on `swift-docc-plugin`.

Reviewer 2: Feel a little weird for me. IMO, `swift-docc` developer should make direct call to `docc` instead of relying on `swift-docc-plugin`.

eg. `docc convert/preview xx --additional-symbol-graph-dir xx`

>      [`swift-docc-plugin`](https://github.com/apple-swift-docc-plugin) is a 

Reviewer 2: https://github.com/apple/swift-docc-plugin

Author: Oh yikes. Sorry about the typo, and sorry that I've missed your comment ;( 

I cleaned up the language for brevity, and also referred to the plugin as "Swift-DocC Plugin" to be consistent with the rest of the readme.
